### PR TITLE
Make DP3 work with casacore tab64bit branch

### PR DESCRIPTION
**Pull request by aroffringa**
This is a draft branch to test compilation with casacore's tab64bit branch.

It contains a few TODO comments that need to change before merging. Also, it should not be merged before casacore's tab64bit branch is merged, because it won't compile with older versions of casacore.

### CODE REVIEW

File: DPPP/Filter.cc [hunk 1]
```diff
@@ -265,12 +265,12 @@ namespace DP3 {
       // Remove these rows from the ANTENNA table.
       // Note that stations of baselines that have been filtered out before,
       // will also be removed.
-      Vector<uInt> removedAnt = selTab.rowNumbers();
-      Vector<Int> antMap = createIdMap (antTab.nrow(), removedAnt);
+      Vector<rownr_t> removedAnt = selTab.rowNumbers();
+      Vector<rownr_t> antMap = createIdMap (antTab.nrow(), removedAnt);
```
Comment (tammojan) [inline DPPP/Filter.cc:269]: This should not be a `rownr_t`, createIdMap uses a -1 for non-mapped antennas.

File: DPPP/Filter.cc [hunk 2]
```diff
@@ -265,12 +265,12 @@ namespace DP3 {
       // Remove these rows from the ANTENNA table.
       // Note that stations of baselines that have been filtered out before,
       // will also be removed.
-      Vector<uInt> removedAnt = selTab.rowNumbers();
-      Vector<Int> antMap = createIdMap (antTab.nrow(), removedAnt);
+      Vector<rownr_t> removedAnt = selTab.rowNumbers();
```
Comment (tammojan) [inline DPPP/Filter.cc:268]: While this is a `rownr_t`, it maps to the antenna table, I think we can safely assume the number of antennas to fit in an int.

**Comment by tammojan** [comment]
The tests will likely fail because the tab64bit branch of casacore has an old version. The code now checks for casacore version 3.3.0 and typedefs an internal `rownr_t` (in namespaces DPPP, BBS::ParmDB and BBS::SourceDB) to either casacore's `rownr_t` or `unsigned int`.

I've tested that DPPP, linked to 64bit casacore, can read a 32bit MS. The result can be read back with DPPP linking to the current casacore master (so without 64bit rownrs). I have not tested with an actual 64bit MS.

When this PR is merged, I think casacore/casacore#945 can be merged as well.

**Comment by tammojan** [comment]
@aroffringa I've set travis to use the released version of casacore again, so that travis will not fail when the tab64 bit branch get merged.
I think this PR is ready to be merged when Travis passes.

**Comment by aroffringa** [comment]
Great! Nice to have both versions of casacore work again. I'm happy to have it merged as well when Travis passes.

**Pull request merged**